### PR TITLE
fixes v1.2.0

### PR DESCRIPTION
**Pull request by alexey0511**

### CODE REVIEW

File: src/js/components/login/login.js [hunk 1]
```diff
@@ -35,6 +35,11 @@ class LoginPage extends MaharaBaseComponent {
     backButton = (e) => {
         Router.navigate(PAGE_URL.SERVER);
     }
+    handlePasswordSubmit = (e) => {
+        if(e.charCode == 13) {
+            this.nextButton();
```
Comment (jenofdoom) [inline src/js/components/login/login.js]: Does this need to be bound in the class setup?

File: src/js/components/pending/pending.js [hunk 2]
```diff
@@ -3,8 +3,9 @@ import React from 'react';
 import MaharaBaseComponent from '../base.js';
 import ExpandCollapse      from '../expand-collapse/expand-collapse.js';
 import StateStore          from '../../state.js';
-import {PENDING}           from '../../constants.js';
+import {PENDING, PAGE_URL}           from '../../constants.js';
```
Comment (jenofdoom) [inline src/js/components/pending/pending.js]: would be good to line up the whitespace on this

File: src/js/components/user/user.js [hunk 3]
```diff
@@ -45,8 +45,8 @@ class User extends MaharaBaseComponent {
             icon = defaultIcon;
             displayName = "";
         }
-
-
+        // add timestamp to query to prevent image caching, because we're using same name 
+        icon += `?${Date.now() || ''}`;
```
Comment (jenofdoom) [inline src/js/components/user/user.js:49]: is there anything else we can use that isn't so specific? this means the image will never be cached - if the version string is available that could be better?

**Review by jenofdoom — COMMENTED** [review]
A few comments

**Review by jenofdoom — COMMENTED** [review]
Commit name for the last commit isn't great, if it is just part of the prev commit then do an amend commit and force push

File: src/js/components/user/user.js (hunk 3 above)
Comment (alexey0511) [reply src/js/components/user/user.js:49]: This is user specific information. I'm affraid the only what we can rely on is a timestamp

File: src/js/components/pending/pending.js (hunk 2 above)
Comment (alexey0511) [reply src/js/components/pending/pending.js]: done

File: src/js/components/user/user.js (hunk 3 above)
Comment (alexey0511) [inline src/js/components/user/user.js]: It's user specific info. I'm afraid the most reliable thing would be the timestamp

File: src/js/components/login/login.js (hunk 1 above)
Comment (alexey0511) [reply src/js/components/login/login.js]: What would be better way to do it? ( it hasn't been implemented using form )

File: src/js/components/user/user.js (hunk 3 above)
Comment (agwells) [reply src/js/components/user/user.js:49]: I agree with Jen, the render() method isn't the best place to do this. It means that every time the component re-renders, the image will refresh. It also prevents this component from being "pure" (always producing the same output given the same input), which can hurt React performance.

The user profile icon is actually downloaded into local storage, and the URL we're using here points to the file in local storage. This happens in the "refreshUserIcon" method, in get-sync-data.js. So, you could add a timestamp to the URL generated there, on line 118 of that file.

File: src/js/components/login/login.js (hunk 1 above)
Comment (agwells) [reply src/js/components/login/login.js]: I think the correct way to do it, would be to surround this with a &lt;form> tag, and make its onsubmit function be this.nextButton.

https://facebook.github.io/react/docs/forms.html

File: src/js/components/server/server.js [hunk 4]
```diff
@@ -35,6 +42,12 @@ export default class ServerPage extends MaharaBaseComponent {
     this.setState({serverUrl: this.refs.serverUrl.value});
   }
 
+  handleUrlSubmit = (e) => {
```
Comment (agwells) [inline src/js/components/server/server.js]: Same here as in login.js. We should use a &lt;form> tag, with an onsubmit.

**Review by agwells — CHANGES_REQUESTED** [review]

**Pull request merged**